### PR TITLE
Improve dbus integration

### PR DESCRIPTION
**Pull request by jreidinger**
I do some tasks mentioned in trello card. I introduce rubocop to keep same coding style like rest of yast2 and fix reduce logging tasks

### CODE REVIEW

File: src/include/snapper/dialogs.rb [hunk 1]
```diff
@@ -25,12 +25,9 @@
 # Authors:	Jiri Suchomel <jsuchome@suse.cz>
 
 module Yast
-
```
Comment (aschnell) [inline src/include/snapper/dialogs.rb:28]: What's so bad about this empty line?

File: src/include/snapper/dialogs.rb [hunk 2]
```diff
@@ -25,12 +25,9 @@
...
   module SnapperDialogsInclude
-
```
Comment (aschnell) [inline src/include/snapper/dialogs.rb:30]: And this?

File: src/include/snapper/dialogs.rb [hunk 3]
```diff
@@ -148,19 +138,19 @@ def ModifySnapshotPopup(snapshot)
       if type != :POST
         cont = VBox(
           # popup label, %{num} is number
-          Label(_("Modify Snapshot %{num}") % { :num => num }),
+          Label(format(_("Modify Snapshot %{num}"), num: num)),
```
Comment (aschnell) [inline src/include/snapper/dialogs.rb]: Why is format better than the string function?

File: src/include/snapper/dialogs.rb (hunk 1 above)
Comment (jreidinger) [inline src/include/snapper/dialogs.rb:28]: it is just matter of style and rubocop goal is to have unified style across files. It is similar question if there should be space after `(` or not. For me empty lines make sense in code where it provide additional separation of logic parts of method, but empty line after module definition do not provide additional information.
According to ruby style guide empty lines are used for separation of logic part and separation of individual methods - https://github.com/bbatsov/ruby-style-guide#empty-lines-between-methods
I am fine if we agree that we want to have space after module or class definition, but we should be consistent, so modify shared yast style guide.

File: src/include/snapper/dialogs.rb (hunk 2 above)
Comment (jreidinger) [inline src/include/snapper/dialogs.rb:30]: similar answer as above

File: src/modules/Snapper.rb [hunk 4]
```diff
@@ -604,22 +553,19 @@ def RestoreFiles(snapshot_num, files)
...
-
     # convert string with userdata to a map
     # "a=1, b=2" -> { "a" => "1", "b" => "2" }
     def string_to_userdata(string)
-
       tmp1 = string.split(",")
 
       tmp2 = tmp1.map do |s|
-        s.split("=", 2).map { |t| t.strip }
+        s.split("=", 2).map(&:strip)
```
Comment (aschnell) [inline src/modules/Snapper.rb]: Please comment strange code. At least I don't know what this is.

File: src/include/snapper/dialogs.rb (hunk 3 above)
Comment (jreidinger) [inline src/include/snapper/dialogs.rb]: rubocop is not about better, it is about consistency. So it use one method for same operation. Ruby style guide suggest it https://github.com/bbatsov/ruby-style-guide#sprintf

File: src/modules/Snapper.rb (hunk 4 above)
Comment (jreidinger) [inline src/modules/Snapper.rb]: you mean `&:` construct or generic idea of this line that removes spaces around splitted elements?

Comment (aschnell) [inline src/modules/Snapper.rb]: Yes, I mean the "&:" construct.

File: src/include/snapper/dialogs.rb (hunk 1 above)
Comment (aschnell) [inline src/include/snapper/dialogs.rb:28]: Who the hell is bbatsov? We use https://github.com/SUSE/style-guides/blob/master/Ruby.md, see https://github.com/yast/yast.github.io/blob/master/doc/code-review.md.

Comment (jreidinger) [inline src/include/snapper/dialogs.rb:28]: bbatsov is autohor of rubocop and his style-guide is semi official ruby style guide. I see that our style guide do not mention it, but when we discuss it conclusion was that we follow this widely known guide and mention only our specific extension to style guide. I will open pull request for our style guide to reflect it.

File: src/modules/Snapper.rb (hunk 4 above)
Comment (jreidinger) [inline src/modules/Snapper.rb]: It is in ruby from 1.9. I can document it everywhere it is used, but it is part of standard language, so I am not sure if it make sense. For c++ you also do not comment all features from new standard and expect that reader know it.

Comment (aschnell) [inline src/modules/Snapper.rb]: OK, don't document it. I'll remember that documentation is not needed when I use C++17 features.

**Comment by aschnell** [comment]
General problem with applying coding styles in this branch is that it will have to be reviewed twice.

**Comment by jreidinger** [comment]
OK, so I can split it and make rubocop support separated pr to yast/yast-snapper and have in this one only reduced logging and tests + leading `/`. We just then need to solve conflicts, but it won't be so big I hope.

**Comment by aschnell** [comment]
To avoid merge conflicts it would be better to postpone the rubocup stuff until the main task is done.

**Comment by jreidinger** [comment]
@aschnell reworked, so now it do not contain rubocop changes

**Pull request merged**